kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -0,0 +1,74 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+-}
+
+module Kore.Step.Rule.Simplify
+    ( simplifyOnePathRuleLhs
+    ) where
+
+import qualified Branch as BranchT
+    ( gather
+    , scatter
+    )
+import Kore.Internal.Conditional
+    ( Conditional (Conditional)
+    )
+import qualified Kore.Internal.Conditional as Conditional.DoNotUse
+    ( Conditional (..)
+    )
+import qualified Kore.Internal.Conditional as Conditional
+    ( withCondition
+    )
+import qualified Kore.Internal.Predicate as Predicate
+    ( fromPredicate
+    )
+import Kore.Step.Rule
+    ( OnePathRule (..)
+    , RulePattern (RulePattern)
+    )
+import qualified Kore.Step.Rule as RulePattern
+    ( RulePattern (..)
+    , fullyApplySubstitution
+    )
+import qualified Kore.Step.Simplification.Pattern as Pattern
+    ( simplifyAndRemoveTopExists
+    )
+import Kore.Step.Simplification.Simplify
+    ( MonadSimplify
+    , SimplifierVariable
+    )
+
+simplifyOnePathRuleLhs
+    ::  ( MonadSimplify simplifier
+        , SimplifierVariable variable
+        )
+    => OnePathRule variable
+    -> simplifier [OnePathRule variable]
+simplifyOnePathRuleLhs (OnePathRule rule) = do
+    simplifiedList <- simplifyRuleLhs rule
+    return (map OnePathRule simplifiedList)
+
+simplifyRuleLhs
+    ::  ( MonadSimplify simplifier
+        , SimplifierVariable variable
+        )
+    => RulePattern variable
+    -> simplifier [RulePattern variable]
+simplifyRuleLhs rule@(RulePattern _ _ _ _ _ _) = do
+    simplified <- Pattern.simplifyAndRemoveTopExists
+        (left `Conditional.withCondition` Predicate.fromPredicate requires)
+    BranchT.gather $ do
```

Comment:
This `scatter`/`gather` combination is just a `map`.

Reply:
Fixed.